sdk/servicebus/Azure.Messaging.ServiceBus/src/ServiceBusMessage.cs

```diff
@@ -108,6 +108,26 @@ public ServiceBusMessage(ServiceBusReceivedMessage receivedMessage)
         /// </remarks>
         public BinaryData Body { get; set; }
 
+        internal enum AmqpBodyType
+        {
+            Unspecified,
+            Data,
+            Sequence,
+            Value
+        }
+
+        internal AmqpBodyType AmqpBodyTypeHint { get; set; }
```

Comment:
Why do we need both BodyTypeHint and BodyType?

Reply:
I ended up exposing `BodyType` as part of the `AmqpTransportBody` class.